apps/dashboard/app/controllers/files_controller.rb

```diff
@@ -58,22 +58,25 @@ def fs
             response.sending_file = true
             response.cache_control[:public] ||= false
 
-            # FIXME: strategy 1: is below, use zip_tricks
-            # strategy 2: use actual zip command (likely much faster) and ActionController::Live
-            zip_tricks_stream do |zip|
-              @path.files_to_zip.each do |file|
-                begin
-                  next unless File.readable?(file.realpath)
-
-                  if File.file?(file.realpath)
-                    zip.write_deflated_file(file.relative_path.to_s) do |zip_file|
-                      IO.copy_stream(file.realpath, zip_file)
+            zip_headers = ZipKit::OutputEnumerator.new.streaming_http_headers
+            response.headers.merge!(zip_headers)
+
+            send_stream(filename: zipname) do |stream|
```

Comment:
tests are failing because `send_stream` is a rails 7 helper.  I think if we just define it ourselves in 3.1 we'll be OK.

Here it is pulled from rails:
```ruby
    def send_stream(filename:, disposition: "attachment", type: nil)
      response.headers["Content-Type"] =
        (type.is_a?(Symbol) ? Mime[type].to_s : type) ||
        Mime::Type.lookup_by_extension(File.extname(filename).downcase.delete("."))&.to_s ||
        "application/octet-stream"

      response.headers["Content-Disposition"] =
        ActionDispatch::Http::ContentDisposition.format(disposition: disposition, filename: filename)

      yield response.stream
    ensure
      response.stream.close
    end
```